### PR TITLE
Fix Captain's Hook Unattach triggering and sacrificing even if it gets destroyed before Equip ability resolves

### PR DESCRIPTION
**Pull request by tool4ever**

### CODE REVIEW

File: forge-game/src/main/java/forge/game/ability/effects/DrainManaEffect.java [hunk 1]
```diff
@@ -23,13 +30,27 @@ protected String getStackDescription(SpellAbility sa) {
...
+        final Map<Player, Integer> lossMap = Maps.newHashMap();
 
         for (final Player p : getTargetPlayers(sa)) {
             if (!p.isInGame()) {
                 continue;
             }
-            drained.addAll(p.getManaPool().clearPool(false));
+            List<Mana> cleared = p.getManaPool().clearPool(false);
+            drained.addAll(cleared);
+            if (game.getStaticEffects().getGlobalRuleChange(GlobalRuleChange.manaBurn)) {
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/ability/effects/DrainManaEffect.java:43]: use ManaPool.hasBurn instead

or better, if possible remove the GobalRuleChange

File: forge-game/src/main/java/forge/game/GameEntity.java [hunk 2]
```diff
@@ -232,6 +232,14 @@ public boolean canBeAttached(final Card attach, SpellAbility sa, boolean checkSB
             return false;
         }
 
+        final Card gameCard = attach.getGame().getCardState(attach, null);
+        // gameCard is LKI in that case, the card is not in game anymore
+        // or the timestamp did change
+        // this should check Self too
+        if (gameCard == null || !attach.equalsWithTimestamp(gameCard)) {
+            return false;
+        }
```
[the remark this answers is not in the document]
Comment (tool4ever) [reply forge-game/src/main/java/forge/game/GameEntity.java:241]: Hmn I tested
- AttachedTo with _Accursed Witch_
- and indirect with _Cosmic Intervention_

File: forge-game/src/main/java/forge/game/ability/effects/DrainManaEffect.java [hunk 3]
```diff
@@ -23,13 +30,27 @@ protected String getStackDescription(SpellAbility sa) {
...
+        final Map<Player, Integer> lossMap = Maps.newHashMap();
 
         for (final Player p : getTargetPlayers(sa)) {
             if (!p.isInGame()) {
                 continue;
             }
-            drained.addAll(p.getManaPool().clearPool(false));
+            List<Mana> cleared = p.getManaPool().clearPool(false);
+            drained.addAll(cleared);
+            if (StaticAbilityUnspentMana.hasManaBurn(p)) {
```
[the remark this answers is not in the document]
Comment (tool4ever) [reply forge-game/src/main/java/forge/game/ability/effects/DrainManaEffect.java:43]: the rules field is set by preferences
imo that's supposed to simulate the old mana burn when a phase ends
so it should not be checked by "lose unspent by effect"

**Review by Hanmac — APPROVED** [review]
approved for now

File: forge-game/src/main/java/forge/game/GameEntity.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tool4ever) [reply forge-game/src/main/java/forge/game/GameEntity.java:241]: Ugh it does break for Role tokens :/
I will check if I should just move it inside AttachEffect

**Review by Hanmac — APPROVED** [review]

**Pull request merged**